src/Symfony/Component/Console/Output/ConsoleOutput.php

```diff
@@ -100,14 +101,32 @@ public function setErrorOutput(OutputInterface $error)
      * Returns true if current environment supports writing console output to
      * STDOUT.
      *
-     * IBM iSeries (OS400) exhibits character-encoding issues when writing to
-     * STDOUT and doesn't properly convert ASCII to EBCDIC, resulting in garbage
-     * output.
-     *
      * @return bool
      */
     protected function hasStdoutSupport()
     {
-        return ('OS400' != php_uname('s'));
+        return false === $this->runningOS400();
+    }
+
+    /**
+     * Returns true if current environment supports writing console output to
+     * STDERR.
+     *
+     * @return bool
+     */
+    protected function hasStderrSupport()
```

Comment:
It sounds like `hasStdoutSupport` and `hasStderrSupport` could be abstracted away to objects that contain more information like this. I don't think those checks are supposed to be part of this class in the first place.